### ISSUE TITLE
Paypal's 2016 merchant security upgrades

### ISSUE THREAD

**Issue by wikimas** [WadeShuler/PHP-PayPal-IPN#22]
Hi,

I got this email from PayPal (below) and I'm still using @Quixotix' [PHP-PayPal-IPN](https://github.com/Quixotix/PHP-PayPal-IPN), v.2.1.0.

Can I just upgrade to the new version of IpnListener.php, replace my certs, or is there more to it?

Thanks!

> 2016 merchant security upgrades.
> 
> We recently announced several security upgrades planned for this year, some of which may require you to make changes to your integration. You're receiving this email because we've identified areas of your integration that may need to be upgraded. 
> 
> _What you're about to read is very technical in nature - we understand that. Please contact the parties responsible for your PayPal integration, or your third party vendor (for example, shopping cart provider, and so on) to review this email. They're best positioned to help you make the changes outlined in this email and in the [2016 Merchant Security Roadmap Microsite](https://www.paypal-knowledge.com/infocenter/index?page=content&id=FAQ1913&expand=true&locale=en_US)._
> 
> WHAT DO I NEED TO DO TO AS A MERCHANT? 
> 
> Here are the steps you'll need to take to ensure your integration is up to date and you don't experience a disruption of service when the changes happen. 
> 
> STEP 1: CONSULT WITH SOMEONE WHO UNDERSTANDS YOUR INTEGRATION. We encourage you to consult with the parties that set up your integration, which could be a consultant or third-party shopping cart. You may also need to find someone who can assist with making your integration changes. 
> 
> STEP 2: UNDERSTAND HOW THESE CHANGES AFFECT YOUR INTEGRATION. Here are the
> key areas requiring your attention. 
> 
> - If the chart shows "Yes", you may require changes to be compatible with that security upgrade.
> - If you see a "No," our data shows that you are already compliant or do not use that functionality. 
> 
> There may be other changes you need to make, but please pay particular attention to the following areas: 
> 
> CHANGE
> DO I NEED TO MAKE A CHANGE?
> 
> [SSL Certificate Upgrade to SHA-256](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1766&viewlocale=en_US)  		**Yes**
> [TLS 1.2 and HTTP/1.1 Upgrade](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1914&viewlocale=en_US)  		**Yes**
> [IPN Verification Postback to HTTPS](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1916&viewlocale=en_US)  		**No**
> [IP Address Update for PayPal Secure FTP Servers](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1912&viewlocale=en_US)  		**No**
> [Merchant API Certificate Credential Upgrade](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1915&viewlocale=en_US)  		**No**
> [Discontinue Use of GET Method for Classic NVP/SOAP APIs](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1927&viewlocale=en_US)  		**No**
> 
> STEP 3: GET THE TECHNICAL DETAILS ABOUT THESE CHANGES. Detailed information about each of the changes and a location to test your integration are available on our [2016 Merchant Security Roadmap Microsite](https://www.paypal-knowledge.com/infocenter/index?page=content&id=FAQ1913&expand=true&locale=en_US). Select the hyperlinks in the chart for information about specific change events. 
> 
> STEP 4: MAKE THE APPROPRIATE CHANGES BY EACH “ACT BY” DATE*. It's important to have your changes in place by the "Act by" date for each change event. 
> 
> STEP 5: FUTURE-PROOF YOUR INTEGRATION. We recommend that you go through the Best Practices section on our [2016 Merchant Security Roadmap Microsite](https://www.paypal-knowledge.com/infocenter/index?page=content&id=FAQ1913&expand=true&locale=en_US).
> 
> 
> WHY IS PAYPAL MAKING THESE CHANGES? 
> 
> Protecting customer information is PayPal's top priority. We support industry standards, such as crypto-industry's mandate to upgrade SSL certificates to SHA-256, and Payment Card Industry (PCI) Council's TLS 1.2 mandate. We also surpass those standards by investing in and building some of the finest protection available. By addressing these changes this year, we believe it helps future-proof your integration and reduce the need to invest in changing your integration in the near future.  
> 
> If you have any questions, visit our Help Centre by clicking HELP on any PayPal page. If you require further assistance, please call us on 805 542 713. 
> 
> Thank you for your support of our commitment to maintain the highest security standards for all our global customers. 
> 
> *Scheduled change dates, including "Act by" dates, provided in this email and the PayPal 2016 Merchant Security Roadmap are subject to change. You'll be notified immediately of any changes to these plans.
> 
> WAS THIS EMAIL HELPFUL? PLEASE CLICK [HERE](https://www.surveymonkey.com/r/PPcustomers) TO LET US KNOW HOW WE'RE DOING AT KEEPING YOU INFORMED.

**Comment by RKO1195** [comment]
Yes me and a friend have both got them letters from PayPal. Mine works still but he changed to a new server and his is giving the curl error here: 


 cURL error: [60] SSL certificate problem: unable to get local issuer certificate
